### ISSUE TITLE
Could not decode AMQP Frame

### ISSUE THREAD

**Issue by Sharad-Regoti-Unotech** [Azure/go-amqp#29]
My code 
```
type activeMQManager struct {
	amqSender *amqp.Sender
}

func initActiveMq(ctx context.Context, logger *model.Logger) (*activeMQManager, error) {
	// Try to connect with AMQ
	client, err := amqp.Dial(logger.Conn)
	if err != nil {
		return nil, err
	}

	
	// Open a session
	amqSession, err := client.NewSession()
	if err != nil {
		return nil, err
	}

	// Create a sender
	sender, err := amqSession.NewSender(
		amqp.LinkTargetAddress(logger.Topic),
	)
	if err != nil {
		return nil, err
	}
	fmt.Println("Connected to active mq")

	return &activeMQManager{amqSender: sender}, nil
}

// LogRequest logs headers,URL,body of the request to AMQ
func (m *activeMQManager) LogRequest(ctx context.Context, r *http.Request) error {
	// Set a request id in header if it doesn't exist
	requestID := r.Header.Get(model.RequestIDHeader)
	if requestID == "" {
		// set a new request id header of request
		requestID = ksuid.New().String()
		r.Header.Add(model.RequestIDHeader, requestID)
	}

	// copy request body
	reqBody, _ := ioutil.ReadAll(r.Body)
	r.Body = ioutil.NopCloser(bytes.NewBuffer(reqBody))
	urlParams, _ := url.ParseQuery(r.URL.RawQuery)
	payload := utils.M{
		"type":        "request",
		"requestId":   requestID,
		"headers":     r.Header,
		"ts":          time.Now().UTC(),
		"url":         r.URL.String(),
		"urlPath":     r.URL.Path,
		"urlParams":   urlParams,
		"urlFragment": r.URL.Fragment,
		"body":        string(reqBody),
	}
	return m.sendAMQMessage(ctx, payload)
}

// LogResponse logs headers,status of the response to AMQ
func (m *activeMQManager) LogResponse(ctx context.Context, requestID string, status int, headers http.Header) error {
	payload := utils.M{
		"type":      "response",
		"ts":        time.Now().UTC(),
		"requestId": requestID,
		"headers":   headers,
		"status":    status,
	}

	return m.sendAMQMessage(ctx, payload)
}

func (m *activeMQManager) sendAMQMessage(ctx context.Context, payload utils.M) error {
	data, err := json.Marshal(payload)
	if err != nil {
		return utils.NewError("Unable to marshal request payload", err)
	}

	// Send message
	return m.amqSender.Send(ctx, amqp.NewMessage(data))
}

```

Client Version: v0.13.6

Error photo
Unable to log response to activemq
![image](https://user-images.githubusercontent.com/82866958/116583650-33f4b200-a934-11eb-83fa-80275d060ffe.png)

**Comment by Sharad-Regoti-Unotech** [comment]
After debugging I found that whenever an HTTP client abruptly disconnects, the request context gets cancelled.
And while calling `LogReqest` & `LogResponse` i used to pass the request context. Whenever the context get cancelled while writing into ActiveMQ, the above error happens

And this error continues for further requests.  

I was able to solve this by using a separate context of 5 seconds in`sendAMQMessage` instead of taking it from the parent function. also, I have removed the context from the parent function.
